src/models/predicate/mod.rs

```diff
@@ -58,22 +61,46 @@ impl Serialize for PredicateVersion {
     }
 }
 
-impl<'de> Deserialize<'de> for PredicateVersion {
+impl<'de> Deserialize<'de> for PredicateVer {
     fn deserialize<D: Deserializer<'de>>(de: D) -> ::std::result::Result<Self, D::Error> {
         let target: String = Deserialize::deserialize(de)?;
-        PredicateVersion::try_from(target).map_err(|e| DeserializeError::custom(format!("{:?}", e)))
+        PredicateVer::try_from(target).map_err(|e| DeserializeError::custom(format!("{:?}", e)))
     }
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[derive(Debug, PartialEq, Eq, Clone, Serialize)]
 #[serde(untagged)]
 pub enum PredicateWrapper {
     LinkV0_2(LinkV02),
     SLSAProvenanceV0_1(SLSAProvenanceV01),
     SLSAProvenanceV0_2(SLSAProvenanceV02),
 }
 
+impl<'de> Deserialize<'de> for PredicateWrapper {
```

Comment:
Deserialize use `serde_json` deserialize to `value` and then `try_from_value` to auto dispatch.